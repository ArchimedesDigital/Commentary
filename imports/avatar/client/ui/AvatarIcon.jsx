export default function AvatarIcon(props) {
	let url = '/images/default_user.jpg';

	if (props.avatar && props.avatar.url) {
		url = props.avatar.url;
	} else if (props.defaultUrl) {
		url = prop.defaultUrl;
	}

	return <img src={url} alt="icon" />;
}

AvatarIcon.propTypes = {
	defaultUrl: React.PropTypes.string,
	avatar: React.PropTypes.object,
};
