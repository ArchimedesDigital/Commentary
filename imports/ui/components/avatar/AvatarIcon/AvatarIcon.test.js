import React from 'react';
import renderer from 'react-test-renderer';

// component:
import AvatarIcon from './AvatarIcon';

describe('OAuthButtons', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<AvatarIcon />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
