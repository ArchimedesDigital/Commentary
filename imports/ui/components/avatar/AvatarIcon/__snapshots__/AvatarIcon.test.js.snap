// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OAuthButtons renders correctly 1`] = `
<div
  alt="icon"
  className="avatar-icon"
  style={
    Object {
      "backgroundImage": "url(/images/default_user.jpg)",
    }
  }
/>
`;
